### ISSUE TITLE
py-pillow build error on Orion with gnu-10.2.0 (`gcc: error: unrecognized command-line option '-R'`) - update py-setuptools to 59.4.0 or remove version 57.4.0 from common packages.yaml

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#206]
I think this error comes from using an external and very old `m4`. Trying to remove the external `m4` from the orion config and build again.

**If** that works, then also try without the external `m4` with Intel. (The intel build didn't have any problem with it.)

**Comment by climbfuji** [comment]
I figured that the problem goes away if upgrading `py-setuptools` from `57.4.0` to `59.4.0`, because the default option for `rpath` linking in `59.4.0` is `-Wl,-R` as opposed to just `-R` in `57.4.0`.

I think that there is still a bug somewhere and that garbage like `compiler_compat` is passed to routine `is_gcc` in `py-setuptools/_distutils/unixccompiler.py`, but at least it doesn’t hurt with `59.4.0`.

I will try to bump the `py-setuptools` version in the common `packages.yaml` to 59.4.0 (or just remove it). I remember we had to pin `py-setuptools` to 57.4.0 in the early days of spack-stack due to multiple versions of one package being built. But maybe that is no longer needed.
